GitHubData/Factories/HTTPClientFactory.cs

```diff
@@ -0,0 +1,30 @@
+﻿// <copyright file="HTTPClientFactory.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+namespace GitHubData.Factories;
+
+/// <summary>
+/// Creates an <see cref="IHttpClient"/> for making HTTP request.
+/// </summary>
+internal static class HTTPClientFactory
+{
+    private static IHttpClient? httpClientSingleton;
+
+    /// <summary>
+    /// Creates an new <see cref="IHttpClient"/> instance or returns the same <see cref="IHttpClient"/> singleton.
+    /// </summary>
+    /// <param name="useAsSingleton">If <c>true</c>, will return the same instance every time.</param>
+    /// <returns>
+    ///     A new or singleton <see cref="IHttpClient"/> instance.
+    /// </returns>
+    public static IHttpClient CreateHTTPClient(bool useAsSingleton = false)
```

Comment:
Refactor name of argument to `createAsSingleton`

```suggestion
    public static IHttpClient CreateHTTPClient(bool createAsSingleton = false)
```